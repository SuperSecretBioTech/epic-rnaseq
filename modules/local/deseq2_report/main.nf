process DESEQ2_REPORT {
    label "process_medium"

    // (Bio)conda packages have intentionally not been pinned to a specific version
    // This was to avoid the pipeline failing due to package conflicts whilst creating the environment when using -profile conda
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-8849acf39a43cdd6c839a369a74c0adc823e2f91:ab110436faf952a33575c64dd74615a84011450b-0' :
        'biocontainers/mulled-v2-8849acf39a43cdd6c839a369a74c0adc823e2f91:ab110436faf952a33575c64dd74615a84011450b-0' }"

    input:
    val study
    path samplesheet
    path salmon

    output:
    path "*.html"
    path "*.deg"

    when:
    task.ext.when == null || task.ext.when

    script:
    """
    Rscript -e "rmarkdown::render(
        "bin/deseq2.rmd",
        params = list(
            study = "$study", 
            metadata_path = "$samplesheet",
            salmon_path = "$salmon"
        )
    )"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        bioconductor-deseq2: \$(Rscript -e "library(DESeq2); cat(as.character(packageVersion('DESeq2')))")
    END_VERSIONS
    """
}