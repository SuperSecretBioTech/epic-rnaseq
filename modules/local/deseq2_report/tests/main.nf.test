nextflow_process {

    name "Test Process DESEQ2_REPORT"
    script "modules/local/deseq2_report/main.nf"
    process "DESEQ2_REPORT"

    test("deseq2 [html]") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = "JIRA_BDS-1008_Mouse_3m"
                input[1] = file("/home/ec2-user/projects/epic_rnaseq/epic-rnaseq/mouse_3m_samplesheet_deseq2.csv")
                input[2] = files("/home/ec2-user/projects/epic_rnaseq/epic-rnaseq/data/")

                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
