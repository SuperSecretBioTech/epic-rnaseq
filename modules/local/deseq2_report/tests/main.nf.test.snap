{
    "deseq2 [html]": {
        "content": [
            {
                "0": [
                    "test.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "test.deg:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-10-15T21:44:19.190630072"
    }
}